File: SecondYear/semestrul4/MPP/PROIECT/CSharpProjectNetworking/Server/service/CategorieVarstaService.cs
using Model;
using  Persistence.database;

namespace Server;
public class CategorieVarstaService : AbstractService<int, CategorieVarsta>
{
    private readonly CategorieVarstaRepository categorieVarstaRepository;

    public CategorieVarstaService(CategorieVarstaRepository repository) : base(repository)
    {
        this.categorieVarstaRepository = repository;
    }

    public CategorieVarsta FindVarstaByRange(string selectedCategory)
    {
        return categorieVarstaRepository.FindVarstaByRange(selectedCategory);
    }
}

File: SecondYear/semestrul4/MPP/PROIECT/JavaProjectORM/ProjectServer/src/main/java/ro/mpp2025/javaprojectui/service/CategorieVarstaService.java
package ro.mpp2025.javaprojectui.service;


import ro.mpp2025.javaprojectui.CategorieVarsta;
import ro.mpp2025.javaprojectui.database.CategorieVarstaRepository;

public class CategorieVarstaService extends AbstractService<Integer, CategorieVarsta> {
    private final CategorieVarstaRepository categorieVarstaRepository;
    public CategorieVarstaService(CategorieVarstaRepository repository) {
        super(repository);
        this.categorieVarstaRepository = repository;
    }

    public CategorieVarsta findVarstaByRange(String selectedCategory) {
        return categorieVarstaRepository.findVarstaByRange(selectedCategory);
    }
}
